toolchain/parser/parse_node_kind.def

```diff
@@ -285,6 +285,13 @@ CARBON_PARSE_NODE_KIND_CHILD_COUNT(InterfaceBodyStart, 0)
 CARBON_PARSE_NODE_KIND_BRACKET(InterfaceBody, InterfaceBodyStart)
 CARBON_PARSE_NODE_KIND_BRACKET(InterfaceDefinition, DeclaredName)
 
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(SelfDeducedParameter, 0)
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(SelfDeducedParameterAddress, 1)
+CARBON_PARSE_NODE_KIND_CHILD_COUNT(Address, 0)
```

Comment:
Structurally, I think when there's `addr`, it should probably have the pattern binding (in full) as the child. i.e., with `CARBON_PARSE_NODE_KIND_CHILD_COUNT(Address, 1)`:

```
    SelfDeducedParameter
    SelfType
  PatternBinding
Address
```

With that, you don't need SelfDeducedParameterAddress anymore, and it should also be easier to reuse `Address`. Instead, you're just adding a conditional that, if you see `addr` in front, you add it as a parent at the end.